package.json

```diff
@@ -70,6 +70,7 @@
     "stacktrace-gps": "^3.0.4",
     "style-loader": "^2.0.0",
     "terser-webpack-plugin": "^4.1.0",
+    "text-field-edit": "^3.1.0",
```

Comment:
The `text-field-edit` library doesn't seem  super popular (only 176 downloads/week) but seems maintained has a clean API that handles the undo issues here, so seems like a good fit.